starknetup.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/sh
+# shellcheck shell=dash
+
+set -eu
+
+ASDF_INSTALL_DOCS="https://asdf-vm.com/guide/getting-started.html"
+SCARB_UNINSTALL_DOCS="https://docs.swmansion.com/scarb/download#uninstall"
+STARKNET_FOUNDRY_UNINSTALL_DOCS="PENDING"
+SCRIPT_VERSION="0.1.0"
+DEFAULT_ASDF_VERSION="v0.14.1"
+
+usage() {
+	cat <<EOF
+The installer for Starknet tools. Installs the latest versions of Scarb, Starknet Foundry and Universal Sierra Compiler using asdf.
+
+Usage: $0 [OPTIONS]
+
+Options:
+  -h, --help      Print help
+  -V, --version   Print script version
+
+EOF
+}
+
+main() {
+	for arg in "$@"; do
+		case "$arg" in
+		-h | --help)
+			usage
+			exit 0
+			;;
+		-V | --version)
+			printf "starknetup %s\n" "$SCRIPT_VERSION"
+			exit 0
+			;;
+		*)
+			err "invalid option '$arg'. For more information, try '--help'."
+			;;
+		esac
+	done
+
+	assert_dependencies
+	assert_not_installed "scarb" $SCARB_UNINSTALL_DOCS
+	install_latest_asdf_plugin "scarb"
+	install_latest_version "scarb"
+	set_global_latest_version "scarb"
+
+	assert_not_installed "starknet-foundry" $STARKNET_FOUNDRY_UNINSTALL_DOCS
+	install_latest_asdf_plugin "starknet-foundry"
+
+	# Reinstall to ensure the latest version of USC is installed
+	uninstall_latest_version "starknet-foundry"
+	install_latest_version "starknet-foundry"
+	set_global_latest_version "starknet-foundry"
+
+	say "Installation complete"
```

Comment:
It should also instruct the user to run `.bashrc` or restart the shell, like when installing asdf. Right now I am getting: 
```
starknetup: Installation complete
ubuntu@93317c8f31ad:~$ universal-sierra-compiler
bash: universal-sierra-compiler: command not found
```